WeakAuras/AuraEnvironment.lua

```diff
@@ -145,7 +145,23 @@ function WeakAuras.ActivateAuraEnvironment(id, cloneId, state)
       current_aura_env.cloneId = cloneId
       current_aura_env.state = state
       current_aura_env.region = WeakAuras.GetRegion(id, cloneId)
-      current_aura_env.config = CopyTable(data.config)
+      if data.controlledChildren then
+        current_aura_env.configs = {}
+        current_aura_env.child_envs = {}
+        for dataIndex, childID in ipairs(data.controlledChildren) do
+          local childData = WeakAuras.GetData(childID)
+          current_aura_env.configs[dataIndex] = CopyTable(childData.config)
+          if childData then
+            if not environment_initialized[childID] then
+              WeakAuras.ActivateAuraEnvironment(childID)
+              WeakAuras.ActivateAuraEnvironment()
+            end
+            current_aura_env.child_envs[dataIndex] = aura_environments[childID]
+          end
+        end
+      else
+        current_aura_env.config = CopyTable(data.config)
```

Comment:
So, one idea I'm having on reading this code again, is that we might want to use the "read-only" table trick here.

See, e.g. http://lua-users.org/wiki/ReadOnlyTables on how that works. That does break a few things though.

Reply:
I don’t really see the point, honestly.